components/ProfileInfo/ProfileInfo.tsx

```diff
@@ -11,39 +13,253 @@ export type FieldInfo = {
+      }
+      if (phone && phone != admin?.phone) {
+        newData["phone"] = phone;
+      }
+      if (email && email != admin?.email) {
+        newData["email"] = email;
+      }
+      const currAdmin = admin;
+      const res = await fetch("/api/auth/updateAdmin", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify({
+          adminID: currAdmin,
+          adminData: newData,
+        }),
+      });
+      // const updatedAdmin = await getAdmin()
+    } catch (e) {
+      throw new Error(e);
+    } finally {
+      null;
+    }
+  }
+
+  function inLineEdit(field) {
+    if (isEdit && field.fieldValue == currEdit) {
+      return (
+        <div>
+          {/* <TextField
+            hiddenLabel
+            id="filed-hidden-label-small"
+            defaultValue={field.fieldValue}
+            variant="filled"
+            size="small"
+          /> */}
+          <form>
+            <input
+              type="text"
+              value={inputValue}
+              onChange={(e) => setInputValue(e.target.value)}
+              className={styles["edit"]}
+            />
+          </form>
+        </div>
+      );
+    }
+    if (field.fieldValue == currEdit) {
+      return (
+        <div className={styles["editState2"]}>
+          <Typography
+            variant="subtitle2"
+            color="#131313"
+            className={styles["text"]}
+            onClick={() => setIsEdit(true)}
+          >
+            {field.fieldValue}
+          </Typography>
+          <IconButton
+            onClick={() => {
+              setIsSelected(false), setInputValue(field.fieldValue);
+            }}
+          >
+            <Icon type="editingpencil" />
+          </IconButton>
+        </div>
+      );
+    }
+    return (
+      <div className={styles["values"]}>
+        <Typography
+          variant="subtitle2"
+          color="#131313"
+          className={styles["text"]}
+        >
+          {field.fieldValue}
+        </Typography>
+      </div>
+    );
+  }
+
+  function fieldValues(field) {
+    if (
+      isEditing &&
+      (field.fieldName == "NAME" ||
+        field.fieldName == "ROLE" ||
+        field.fieldName == "EMAIL" ||
+        field.fieldName == "PHONE #")
+    ) {
+      if (isSelected) {
+        return inLineEdit(field);
+      }
+      return (
+        <div className={styles["editState1"]}>
+          <Typography
+            variant="subtitle2"
+            color="#131313"
+            className={styles["text"]}
+          >
+            {field.fieldValue}
+          </Typography>
+          <IconButton
+            onClick={() => {
+              setInputValue(field.fieldValue),
+                setIsSelected(true),
+                setCurrEdit(field.fieldValue);
+            }}
+          >
+            <Icon type="editingpencil" />
+          </IconButton>
+        </div>
+      );
+    }
+    return (
```

Comment:
I think it would be better to use forms and inputs to handle this. For fields you aren't actively editing, you could use an input element that's disabled. If you need help implementing this, let me know